electrum/gui/qt/plugins_dialog.py

```diff
@@ -70,63 +121,161 @@ def __init__(self, window: 'ElectrumWindow'):
+        vbox.addLayout(Buttons(add_button, CloseButton(self)))
         self.show_list()
 
-    def enable_settings_widget(self, name: str, i: int):
-        p = self.plugins.get(name)
-        widget = self.settings_widgets.get(name)  # type: Optional[QWidget]
-        if widget and not p:
-            # plugin got disabled, rm widget
-            self.grid.removeWidget(widget)
-            widget.setParent(None)
-            self.settings_widgets.pop(name)
-        elif widget is None and p and p.requires_settings() and p.is_enabled():
-            # plugin got enabled, add widget
-            widget = self.settings_widgets[name] = p.settings_widget(self)
-            self.grid.addWidget(widget, i, 1)
+    def get_plugins_privkey(self) -> Optional['ECPrivkey']:
+        pubkey, salt = self.plugins.get_pubkey_bytes()
+        if not pubkey:
+            self.init_plugins_password()
+            return
+        # ask for url and password, same window
+        pw = self.window.password_dialog(
+            msg=' '.join([
+                _('<b>Warning</b>: Third-party plugins are not endorsed by Electrum!'),
+                '<br/><br/>',
+                _('If you install a third-party plugin, you trust the software not to be malicious.'),
+                _('Electrum will not be responsible in case of theft, loss of funds or privacy that might result from third-party plugins.'),
+                _('You should at minimum check who the author of the plugin is, and be careful of imposters.'),
+                '<br/><br/>',
+                _('Please enter your plugin authorization password') + ':'
+            ])
+        )
+        if not pw:
+            return
+        privkey = self.plugins.derive_privkey(pw, salt)
+        if pubkey != privkey.get_public_key_bytes():
+            keyfile_path, keyfile_help = self.plugins.get_keyfile_path()
+            self.window.show_error(
+                ''.join([
+                    _('Incorrect password.'), '\n\n',
+                    _('Your plugin authorization password is required to install plugins.'), ' ',
+                    _('If you need to reset it, remove the following file:'), '\n\n',
+                    keyfile_path
+                ]))
+            return
+        return privkey
+
+    def init_plugins_password(self):
+        from .password_dialog import NewPasswordDialog
+        import os
+        msg = ' '.join([
+            _('In order to install third-party plugins, you need to choose a plugin authorization password.'),
+            _('Its purpose is to prevent unauthorized users (or malware) from installing plugins.'),
+        ])
+        d = NewPasswordDialog(self, msg=msg)
+        pw = d.run()
+        if not pw:
+            return
+        salt = os.urandom(32)
+        privkey = self.plugins.derive_privkey(pw, salt)
+        pubkey = privkey.get_public_key_bytes()
+        key_hex = salt.hex() + pubkey.hex()
```

Comment:
I think we should version this. That would for example allow changing the KDF in the future.

Just prefix it with e.g. `b"epp1".hex()`
(short for electrum plugin password)

Reply:
done, see `PLUGIN_PASSWORD_VERSION`